bin/yb-ctl

```diff
@@ -1287,23 +1420,40 @@ class ClusterControl:
 
     # Starts as well as creates. Check if the cluster exists.
     # If it does not create it else start the individual daemons.
-    def start(self):
+    #
+    # This should really be called start_cluster.
+    def start_cmd_impl(self):
         if os.path.isdir(self.options.cluster_base_dir):
-            print("Starting cluster.")
+            print("Starting cluster with base directory %s" % self.options.cluster_base_dir)
+
             self.set_master_addresses()
             self.for_all_daemons(self.start_daemon)
             self.wait_for_cluster_or_raise()
             self.modify_placement_info()
+
+            if self.options.enable_ysql and not self.cluster_config.get('enable_ysql'):
+                if self.already_running_daemons:
+                    sys.stderr.write(
+                            "YugaByte DB was already running. If you are trying to enable YSQL "
+                            "on an existing cluster, please stop/start the cluster.\n")
+                    # To prevent YSQL from showing up in the status.
+                    self.options.enable_ysql = False
+                else:
+                    print("YSQL has been enabled on an existing cluster, running initdb")
+                    self.run_cluster_wide_ysql_initdb()
```

Comment:
Lets use `master_auto_run_initdb` while starting the cluster instead of running initdb here?

Reply:
@ndeodhar , @mbautin - since that's the old mechanism we'll retire soon anyway, let's not make those unrelated changes in this diff.

I feel this change is already has too many unrelated things.